src/webapp/components/import-export/ImportTable.tsx

```diff
@@ -0,0 +1,607 @@
+    }, [allUsers, loading]);
+
+    const existingUserInTable = useCallback(
+        (newUsers: User[]) => {
+            if (!existingUsersNames) {
+                return false;
+            }
+            return _(newUsers).some(user => existingUsersNames.includes(user.username));
+        },
+        [existingUsersNames]
+    );
+
+    const closeInfoDialog = () => {
+        setInfoDialog(undefined);
+    };
+
+    const toggleAllowOverwrite = useCallback(
+        (_event, newValue: boolean) => {
+            setAreUsersValid(newValue || !errorsCount);
+            setAllowOverwrite(newValue);
+        },
+        [errorsCount]
+    );
+
+    const renderDialogTitle = () => {
+        const errorText =
+            errorsCount === 0
+                ? null
+                : i18n.t("{{n}} invalid users found, check in-line errors in table", { n: errorsCount });
+        const maxWarnings = 10;
+        const hiddenWarnings = Math.max(warnings.length - maxWarnings, 0);
+
+        const warningText =
+            warnings.length === 0
+                ? null
+                : _([
+                      i18n.t("{{n}} warning(s) while importing file", { n: warnings.length }) + ":",
+                      ..._(warnings)
+                          .take(maxWarnings)
+                          .map((line, idx) => `${idx + 1}. ${line}`)
+                          .value(),
+                      hiddenWarnings > 0 ? i18n.t("and_n_more_warnings", { n: hiddenWarnings }) : null,
+                  ])
+                      .compact()
+                      .join("\n");
+
+        return (
+            <React.Fragment>
+                <StyledDialogTitle>{title}</StyledDialogTitle>
+                {errorText && (
+                    <DialogTooltip title={errorText}>
+                        <FontIcon className="material-icons">error</FontIcon>
+                    </DialogTooltip>
+                )}
+                {warningText && (
+                    <DialogTooltip title={warningText}>
+                        <FontIcon className="material-icons">warning</FontIcon>
+                    </DialogTooltip>
+                )}
+            </React.Fragment>
+        );
+    };
+
+    const onSubmit = useCallback(
+        async ({ users }: { users: User[] }) => {
+            loading.show(true, i18n.t("Importing users"));
+            const { error } = await onSave(users);
+
+            if (error) {
+                setInfoDialog({ response: error });
+                snackbar.error(error);
+            } else {
+                onRequestClose();
+            }
+
+            loading.reset();
+        },
+        [snackbar, loading]
+    );
+
+    const addRow = useCallback(() => {
+        const newUser = {
+            ...defaultUser,
+            id: generateUid(),
+            username: "",
+            password: `District123$`,
+        };
+
+        setUsers(users => users.concat(newUser));
+    }, []);
+
+    let submit: any;
```

Comment:
I know you're doing this because you need to pass the submit method to the `<ConfirmationDialog />`.

But maybe as an alternative we can use the `Dialog` component and then use a submit button to trigger the action.

```tsx
<Dialog open={true} maxWidth={"lg"} fullWidth={true}>
  <TableContainer>
      {/*  add id to the form */}
      <form id="import-form" onSubmit={handleSubmit}>
         ...
      </form>
  </TableContainer>
  {/* we pass the form id to the form attribute */}
  <Button type="submit" form="import-form">
    Import
  </Button>
</Dialog>
```
I know is more code but it's kinda confusing to have a submit variable and reassigned later inside a component. What do you think?

Reply:
You're right, I think it would be clearer and simpler to use material-ui `Dialog`, `ConfirmationDialog` is not the best when using form